rca/wagtailapi/filters.py

```diff
@@ -13,3 +17,50 @@ def filter_queryset(self, request, queryset, view):
             queryset = queryset.filter(degree_level__in=pks)
 
         return queryset
+
+
+class SearchFilter(filters.SearchFilter):
+    def filter_queryset(self, request, queryset, view):
+        """
+        This overrides the wagtail core api filters.SearchFilter
+        so we can provide autocomplet/fuzzy text matching with
+        sb.autocomplete
+        """
+        search_enabled = getattr(settings, "WAGTAILAPI_SEARCH_ENABLED", True)
+
+        if "search" in request.GET:
+            if not search_enabled:
+                raise BadRequestError("search is disabled")
+
+            # Searching and filtering by tag at the same time is not supported
+            if getattr(queryset, "_filtered_by_tag", False):
+                raise BadRequestError(
+                    "filtering by tag with a search query is not supported"
+                )
+
+            search_query = request.GET["search"]
+            search_operator = request.GET.get("search_operator", None)
+            order_by_relevance = "order" not in request.GET
+
+            sb = get_search_backend()
+            try:
+                queryset = sb.autocomplete(
```

Comment:
I am not 100% certain that you need to use `autocomplete` method.

Seems like the generic search method may be able to search through the "auto complete" index too - https://github.com/wagtail/wagtail/blob/28c104ff63e04f1ea4e440a789350f12e79ddc42/wagtail/contrib/postgres_search/backend.py#L302-L304 - but I may be interpreting this wrongly so worth giving it a test.

Reply:
I'm pretty certain, testing a search for `diploma` using 3 letters `dip`:

without sb.autocomplete:

[!img](https://i.imgur.com/pKAkoNM.png)

with sb.autocomplete:

[!img](https://i.imgur.com/SQb53Xz.png)